src/components/vm/nics/nicEdit.jsx

```diff
@@ -19,25 +19,23 @@
 import React from 'react';
 import cockpit from 'cockpit';
 import PropTypes from 'prop-types';
-import { Button, Alert, Form, FormGroup, Modal } from '@patternfly/react-core';
+import { Button, Alert, Form, FormGroup, Modal, TextInput } from '@patternfly/react-core';
 
 import { ModalError } from 'cockpit-components-inline-notification.jsx';
 import { NetworkTypeAndSourceRow, NetworkModelRow } from './nicBody.jsx';
-import {
-    changeNetworkSettings,
-    getVm
-} from '../../../actions/provider-actions.js';
+import { getVm } from '../../../actions/provider-actions.js';
+import { changeNetworkSettings } from '../../../libvirt-dbus.js';
 
 import 'form-layout.scss';
 
 const _ = cockpit.gettext;
 
-const NetworkMacRow = ({ network }) => {
+const NetworkMacRow = ({ mac, onValueChanged, idPrefix }) => {
     return (
-        <FormGroup fieldId='mac' label={_("MAC address")} hasNoPaddingTop>
-            <samp id='mac'>
-                {network.mac}
-            </samp>
+        <FormGroup fieldId="mac" label={_("MAC address")} hasNoPaddingTop>
```

Comment:
The fieldId needs to be the same with the id in the nested component for accessibility purposed.